components/camera/fake/image_file_test.go

```diff
@@ -45,16 +43,14 @@ func TestPCD(t *testing.T) {
 	cam, err = newCamera(ctx, resource.Name{API: camera.API}, cfg, logger)
 	test.That(t, err, test.ShouldBeNil)
 
-	stream, err := cam.Stream(ctx)
-	test.That(t, err, test.ShouldBeNil)
-
 	readInImage, err := rimage.NewImageFromFile(artifact.MustPath("vision/objectdetection/detection_test.jpg"))
 	test.That(t, err, test.ShouldBeNil)
 
-	strmImg, _, err := stream.Next(ctx)
+	imgBytes, _, err := cam.Image(ctx, utils.MimeTypeJPEG, nil)
 	test.That(t, err, test.ShouldBeNil)
-	test.That(t, strmImg, test.ShouldResemble, readInImage)
-	test.That(t, strmImg.Bounds(), test.ShouldResemble, readInImage.Bounds())
+	expectedBytes, err := rimage.EncodeImage(ctx, readInImage, utils.MimeTypeJPEG)
```

Comment:
why not compare file contents, rather than encoding the jpeg?

Reply:
I think my answer to this is the same as https://github.com/viamrobotics/rdk/pull/4487#discussion_r1842882939